classes/Project/Project.mjs

```diff
@@ -42,24 +39,62 @@ export default class Project {
 
   async delete(projectId) {
     if (!projectId) {
-      err_out.message = "Project ID is required"
-      err_out.status = 400
-      throw err_out
+      throw {status: 400, message: "Project ID is required"}
     }
 
     return database.remove(projectId)
   }
+ 
+ checkUserAccess(userAgent) {
+  if (!this.projectData) {
+    return {
+      hasAccess: false,
+      message: "Project data is not loaded.",
+    };
+  } 
 
-  async getById(projectId) {
-    if (!projectId) {
-      err_out.message = "Project ID is required"
-      err_out.status = 400
-      throw err_out
+  if (this.projectData.creator === userAgent) {
```

Comment:
I'm happy to take this in, but we probably would have the agent/user already on the request, since the Project is only returned behind auth.

Reply:
Yes, we have the agent/user on the request, I'm sending that agent from the route to verif against the creator/groupMembers of the project in the Project class